### ISSUE TITLE
License

### ISSUE THREAD

**Issue by sheepo99** [ptitSeb/hydracastlelabyrinth#14]
Hello,

I was interested in submitting this project to the Debian Games Team to include in their main repository. However, because Debian requires confirmation on licensing matters, I would need confirmation Buster did indeed allow you to license the original data files under the GPLv2. If  possible, do you think you can upload the correspondence between you and him?

I was also thinking of contacting Buster himself on this matter. Can you let me know if he is able to communicate well in English?

Kind regards,
Sheepo99

**Comment by ptitSeb** [comment]
Hi,

That's an interresting question, but look in the orinal notice in the readme: "This work has been done by an anon from /hbg/ on 4chan.org/vg/ and not me"

I'm afraid I haven't contacted Buster about this.
What I propose is: you can try to contact him (in english). If english is an issue, tell me, and I'll try to ask a friend taht speak and write japanse to help.

To sum up: the source code is not original code by buster, but an anon person, that put code as GPL2. The data is from buster original game, and hopefully it's ok to distribute (it's freeware as far as I can say).

**Comment by sheepo99** [comment]
In order to distribute the game on the Debian repositores, the data license must be compatible with GPL or Creative Commons, so in this case, being gratis is not enough, and knowing the data license is a must! Do you happen to know where i can find the code repo created by the 4chan dude so I can try and contact him?

One more question do you eventually plan to finish this project, given that at this point it is still in an unfinished state?

**Comment by ptitSeb** [comment]
Ah sorry, I lost track the orignal stuff, but maybe google knows.

I understand the licence issue for debian, and for the Data, only Buster can answer.

For the project, it's pretty much finished now. The game is completable, and suffer no known bugs. I may do some light bug fixing or add some (exotics) plateform support, but appart from that, it's in a completed state.

**Comment by sheepo99** [comment]
Doing a first playthrough I already found some bugs, for instance, there is no options menu in the main menu and it's impossible to access by pressing X unless in game. Another problem is that attempting to close the application window or using alt+F4 resuts in the game either starting/pausing instead!

Finally it would be nice if it would be possible to configure the controls in the options menu,

**Comment by ptitSeb** [comment]
Ah yeah, I should implement Option in the main menu, right.

Quitting the main window lead to Pausing? Using SDL2 build I guess?

Yeah, configuring the controls (including gamepad), would be good, I agree.

**Comment by sheepo99** [comment]
No, I'm using the SDL 1.2 build. Please find attached a video of the issue:

[hcl.mp4.zip](https://github.com/ptitSeb/hydracastlelabyrinth/files/3387333/hcl.mp4.zip)

As you can see when i try to close the window with the mouse it starts/restarts the game. Alt+F4 produces the same results.

**Comment by sheepo99** [comment]
Hello, any updates on this? As you can see I attached a video of the issue.

**Comment by ptitSeb** [comment]
Ah non, sorry, I was busy with other projects. But I'll fix this. I'm pretty sure it's not too complicated, just missing correct handling of `SDL_QUIT` message probably.

**Comment by ptitSeb** [comment]
@sheepo99 : I have pushed something, I think it fixed the issue with the Alt-F4 / Closing the window.

**Comment by sheepo99** [comment]
That fixed it, thanks! If you could also add an the Options menu to the Main Menu and a way to configure keys it would be really nice!

I've also wanted to contact Buster but I do not think he speaks English. If I were to write an email could your friend maybe attempt to translate it?

**Comment by ptitSeb** [comment]
The Option in main menu should be fairly simple. Configuring key is bit more complicated so that one will probably take a bit more time.

For Buster, sure, I can ask if he can translate.

**Comment by ptitSeb** [comment]
So, OPTIONS is in the main menu too now.

**Comment by sheepo99** [comment]
Thank you! You're being a really cool developer by fixing all this stuff so promptly!
I've already written the text to Buster. Let me know to which address I can send it to you for translation. :)

Kind regards

**Comment by ptitSeb** [comment]
Send it to me (my mail should be in your notification mail)

**Comment by sheepo99** [comment]
I cannot see your address in the notification email. 
If you you don't want to reveal your email publicly, you can mail me first at my trash address

**Comment by ptitSeb** [comment]
> I cannot see your address in the notification email. It's all sent via notifications(at)github(dot)com
> 
> If you you don't want to reveal your email publicly, you can mail me first at my trash address 

No I mean, in the message, I had wrote my address, but I then edited it to remove the address. So it should be in the mail you received, but no more apparent. Or maybe you didn't received notification mail, or discard them immediatly?

**Comment by sheepo99** [comment]
Ah, I see it now, thanks! Kindly edit your previous message containing the quote with my address then!

In the meantime I found two bugs concerning character animations and speed. I'll report them later with videos and such.

**Comment by ptitSeb** [comment]
> Ah, I see it now, thanks! Kindly edit your previous message containing the quote with my address then!

Done.

**Comment by sheepo99** [comment]
Hello again,

Did you happen to receive the email I sent you?

**Comment by sheepo99** [comment]
Hello, apart from a reminder to the email that I sent you a while ago I would like to note something:

I've just noticed you seem to have added a copyright note with your name to the main screen. Technically you are not allowed to do this for two reasons:
1 - Until we get confirmation of the opposite, the original game is still copyrighted by Buster, and you cannot add yourself as a copyright holder because you did not create this intellectual property, even if you created the code for this port.
2 - This port and the original code that it is based on is licensed under the GPL, which is incompatible with the notion of copyrighted code. 

As such the correct note that should be added should be:

(C) 2011 E. Hashimoto
GPLv2 2017-2019 ptitSeb

**Comment by ptitSeb** [comment]
Ah, ok, thanks, I'll fix that tomorrow.

Also, about the mail, I had forwarded it to my friend as soon as I received it, I'll ask him if he did something with it.

**Comment by sheepo99** [comment]
Cool, thank you very much! I'm still hoping that we will be able to contact buster sometime. If we ever manage to get the art license changed to GPL we will be able to submit this to the repositories of most major linux distros, where it will be widely disseminated by the community. :) It also means the game will effectively become part of the creative commons which is a very positive thing for public good.

**Comment by ptitSeb** [comment]
> As such the correct note that should be added should be:
> 
> (C) 2011 E. Hashimoto
> GPLv2 2017-2019 ptitSeb

Or should I just drop the (c), and simply have 
```
(C) 2011 E. Hashimoto
2017-2019 ptitSeb
```
I'm quite limited (caps only, not a a lot of space), so I guess things like "Linux & co. port by ptitSeb" are not possible.

**Comment by sheepo99** [comment]
What about this:

(C) 2011 E. Hashimoto
GPLv2 2017 ptitSeb

or if you still lack space

(C) 2011 E. Hashimoto
GPLv2 ptitSeb

Marking something with a date only really matters if you want to set the beginning of a copyright for legal purposes. For GPL and CC licenses the date is not really necessary, but if you want to include it, the initial year of completion for the first stable version should be enough.

**Comment by ptitSeb** [comment]
Ok. Also, I don't have lower case letter, so `GPL2` will work fine, right?

**Comment by sheepo99** [comment]
Yeah that's fine.

**Comment by ptitSeb** [comment]
Ok, I have update the Title screen. You can check the Web version to see the result.
